### PR TITLE
Facebook support in the Go client

### PR DESCRIPTION
**Pull request by oconnor663**
r? @maxtaco 

image

### CODE REVIEW

File: go/externals/proof_support_facebook.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+	splitPath := strings.Split(parsedLink.Path, "/")
+	if len(splitPath) < 1 {
+		return "", libkb.NewProofError(keybase1.ProofStatus_FAILED_PARSE, "Username URL has no path.")
+	}
+	username := splitPath[1]
+	return username, nil
+}
+
+func extractProofText(doc *goquery.Document) (string, libkb.ProofError) {
+	selector := "#m_story_permalink_view h3"
```
Comment (maxtaco) [inline go/externals/proof_support_facebook.go]: that `h3` is a little scary, but....

Comment (oconnor663) [inline go/externals/proof_support_facebook.go]: You're right, I think the current logic can be exploited to make a post that's *not* a proof appear to be a proof, by commenting on it. Let's look together.

**Comment by oconnor663** [comment]
@maxtaco ready for another look.

File: go/externals/proof_support_facebook.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+	// TODO: We could ignore this portion of the server's hint. Should we?
+	if !strings.Contains(h.GetCheckText(), wantedMediumID) {
+		return libkb.NewProofError(keybase1.ProofStatus_BAD_SIGNATURE,
+			"Bad proof-check text from server; need '%s' as a substring", wantedMediumID)
+	}
+
+	return nil
+}
+
+func (rc *FacebookChecker) ScreenNameCompare(s1, s2 string) bool {
```
Comment (maxtaco) [inline go/externals/proof_support_facebook.go]: is this the same as `usernamesEqual`?

File: go/externals/proof_support_facebook.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+	if strings.TrimSpace(proofText) != strings.TrimSpace(h.GetCheckText()) {
+		return libkb.NewProofError(keybase1.ProofStatus_TEXT_NOT_FOUND, "Proof text not found: '%s' != '%s'", proofText, h.GetCheckText())
+	}
+
+	// The proof is good.
+	return nil
+}
+
+func extractUsername(doc *goquery.Document) (string, libkb.ProofError) {
+	selector := "#m_story_permalink_view > div:first-child > div:first-child > div:first-child h3 a"
```
Comment (maxtaco) [inline go/externals/proof_support_facebook.go]: change to grab first `h3` and the `a` inside that `h3`.

**Comment by maxtaco** [comment]
ok!

**Comment by maxtaco** [comment]
👍

**Comment by oconnor663** [comment]
@maxtaco this is ready for a final look. Related changes for the server: https://github.com/keybase/proofs/pull/46

**Comment by maxtaco** [comment]
👍

**Pull request merged**